### DISCUSSION TITLE
How to build/run using other profiles?

### DISCUSSION THREAD

**Discussion by nothingnesses** [90-008/nix-cargo-integration#92] [Q&A] [answered]
Currently, I have a "flake.nix" for my project with contents similar to the following:

```nix
{
  description = "My project's description.";
  inputs = {
    flake-compat = {
      flake = false;
      url = "github:edolstra/flake-compat";
    };
    nci = {
      inputs.nixpkgs.follows = "nixpkgs";
      url = "github:yusdacra/nix-cargo-integration";
    };
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
  };
  outputs = inputs: inputs.nci.lib.makeOutputs {
    pkgConfig = common: {
      package_a = {
        app = true;
        build = true;
      };
    root = ./.;
  };
}
```

Using this, I can run "package_a" in the workspace with `nix run .#package_a`.

Looking more into nci, it seems to be using dream2nix and crane behind the scenes. I read that the latter builds projects using the "release" profile by default. Presumably, nci also uses this profile by default, right? In any case, I'd like to know: how would I build/run my package using other profiles, e.g., with the "debug" profile?

**Comment by 90-008** [comment]
If you do `nix flake show`, you should see `package_a-debug` under `packages` outputs. That will build using the debug profile. The `package_a` one will use the release profile. You can also see the same under `apps` outputs, eg. for a binary there will be `package_a` and `package_a-debug`, one runs the binary build with release and the other runs the binary built with the debug profile.

**Reply by nothingnesses** [reply]
Thanks. Is it also possible to build/run other profiles such as "test", "bench" or any custom profiles defined in Cargo.toml?

**Reply by 90-008** [reply]
Not for bench, but I can implement a feature for it. The `test` profile is exposed under the packages in `checks` outputs, those will use it. Custom profiles would also be possible. If you can make an issue for this, it would be great.

**Comment by 90-008** [comment, marked as answer]
https://github.com/yusdacra/nix-cargo-integration/pull/97 was merged to support this use-case.
